packages/contract-processors/src/moonbeam.ts

```diff
@@ -0,0 +1,278 @@
+// Copyright 2020-2021 OnFinality Limited authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import {Log, TransactionResponse} from '@ethersproject/abstract-provider';
+import {BigNumber} from '@ethersproject/bignumber';
+import {isHexString, hexStripZeros} from '@ethersproject/bytes';
+import {ApiPromise} from '@polkadot/api';
+import {CodecHash} from '@polkadot/types/interfaces';
+import {
+  SubqlDatasourceProcessor,
+  SubqlCustomDatasource,
+  SubqlHandlerKind,
+  SubqlNetworkFilter,
+  SubstrateEvent,
+  SubqlEventFilter,
+  SecondLayerHandlerProcessor,
+  SubstrateExtrinsic,
+} from '@subql/types';
+import {plainToClass} from 'class-transformer';
+import {
+  IsOptional,
+  IsHexadecimal,
+  validateSync,
+  ValidatorConstraint,
+  ValidatorConstraintInterface,
+  Validate,
+  IsEthereumAddress,
+} from 'class-validator';
+
+type TopicFilter = string | string[] | null | undefined;
+
+export type MoonbeamDatasource = SubqlCustomDatasource<'substrate/Moonbeam'>;
+
+export interface MoonbeamEventFilter {
+  address?: string;
+  topics?: [TopicFilter, TopicFilter, TopicFilter, TopicFilter];
+}
+
+export interface MoonbeamCallFilter {
+  from?: string;
+  to?: string;
+  methodId?: string;
+}
+
+export type MoonbeamEvent = Log;
+export type MoonbeamCall = Omit<TransactionResponse, 'wait'>;
```

Comment:
Currently there is some data missing that is present in the [TransactionReceipt](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionReceipt). Not sure if its needed right now or can be added later

Reply:
What's the missing data? I believe we can add them later. Maybe after talking with Moonbeam's team to collect feedbacks from them.